python/tests/transformers/keras_transformer_test.py

```diff
@@ -0,0 +1,145 @@
+        to a dataset of one-dimensional examples.
+        """
+        np.random.seed(self.RANDOM_SEED)
+        return np.random.randn(num_examples, num_features)
+
+    def getInputDF(self, sqlContext, inputCol, idCol):
+        """ Return a DataFrame containing a long ID column and an input column of arrays. """
+        x_train = self._loadNumpyData(num_examples=20, num_features=self.NUM_FEATURES)
+        train_rows = [{idCol : i, inputCol : x_train[i].tolist()} for i in range(len(x_train))]
+        return sqlContext.createDataFrame(train_rows)
+
+    def _getKerasModelWeightInitializer(self):
+        """
+        Get initializer for a set of weights (e.g. the kernel/bias of a single Dense layer)
+        within a Keras model.
+        """
+        return glorot_uniform(seed=self.RANDOM_SEED)
+
+    def getKerasModel(self):
+        """
+        Build and return keras model for (binary) classification on one-dimensional data.
+
+        The model has a single hidden layer that feeds into a single-unit, sigmoid-activated output
+        unit. Weights of the hidden & output layer are drawn from a zero-mean uniform distribution.
+        """
+        model = Sequential()
+
+        # We add a vanilla hidden layer, specifying the initializer (RNG) for the kernel & bias
+        # weights
+        model.add(Dense(units=50, input_shape=(self.NUM_FEATURES, ),
+                        bias_initializer=self._getKerasModelWeightInitializer(),
+                        kernel_initializer=self._getKerasModelWeightInitializer()))
+        model.add(Activation('relu'))
+
+        # We project onto a single unit output layer, and squash it with a sigmoid:
+        model.add(Dense(units=1, bias_initializer=self._getKerasModelWeightInitializer(),
+                        kernel_initializer=self._getKerasModelWeightInitializer()))
+        model.add(Activation('sigmoid'))
+        return model
+
+    def prepareKerasModelFile(self, model, filename):
+        """
+        Saves the passed-in keras model to a temporary directory with the specified filename.
+        """
+        path = os.path.join(self.tempdir, filename)
+        model.save(path)
+        return path
+
+    def executeKerasModel(self, df, model, input_col, id_col):
+        """
+        Given an input DataFrame, applies the passed-in Keras model to the data in the specified
+        input column, returning a numpy array of the model output sorted by increasing row ID (where
+        row ID is contained in id_col).
+        """
+        # Collect dataframe, sort rows by ID column
+        rows = df.select(input_col, id_col).collect()
+        rows.sort(key=lambda row: row[id_col])
+        # Get numpy array (num_examples, num_features) containing input data
+        x_predict = np.array([row[input_col] for row in rows])
+        return model.predict(x_predict)
+
+    def transformOutputToComparables(self, final_df, id_col, output_col):
```

Comment:
maybe `convertToKerasOutputFormat` ? "transform" sounds like we're going to call `.transform`